airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -138,6 +138,9 @@ class KubernetesPodOperator(BaseOperator):  # pylint: disable=too-many-instance-
     :type init_containers: list[kubernetes.client.models.V1Container]
     :param log_events_on_failure: Log the pod's events if a failure occurs
     :type log_events_on_failure: bool
+    :param retry_only_on_pod_launching_failure: retry logic is only effective if pod launching fails. This
```

Comment:
I'm confused here, how does this  know if the image has failed vs. the other arguments (e.g. a failed volume mount?)